src/ucp/api/ucp.h

```diff
@@ -369,19 +369,23 @@ enum ucp_dt_type {
  * ucp_mem_map() function.
  */
 enum {
-    UCP_MEM_MAP_NONBLOCK = UCS_BIT(0), /**< Complete the mapping faster, possibly by
-                                            not populating the pages in the mapping
-                                            up-front, and mapping them later when
-                                            they are accessed by communication
-                                            routines. */
-    UCP_MEM_MAP_ALLOCATE = UCS_BIT(1), /**< Identify requirement for allocation,
-                                            if passed address is not a null-pointer
-                                            then it will be used as a hint or direct
-                                            address for allocation. */
-    UCP_MEM_MAP_FIXED    = UCS_BIT(2)  /**< Don't interpret address as a hint:
-                                            place the mapping at exactly that
-                                            address. The address must be a multiple
-                                            of the page size. */
+    UCP_MEM_MAP_NONBLOCK   = UCS_BIT(0), /**< Complete the mapping faster, possibly by
+                                              not populating the pages in the mapping
+                                              up-front, and mapping them later when
+                                              they are accessed by communication
+                                              routines. */
+    UCP_MEM_MAP_ALLOCATE   = UCS_BIT(1), /**< Identify requirement for allocation,
+                                              if passed address is not a null-pointer
+                                              then it will be used as a hint or direct
+                                              address for allocation. */
+    UCP_MEM_MAP_FIXED      = UCS_BIT(2), /**< Don't interpret address as a hint:
+                                              place the mapping at exactly that
+                                              address. The address must be a multiple
+                                              of the page size. */
+    UCP_MEM_MAP_DEVICE_MEM = UCS_BIT(3)  /**< Allocate memory on device. It allows
+                                              using on-chip memory, located on the
+                                              device, as a data buffer for RDMA operations.
+                                              Allocated buffer can't be accessed directly. */
```

Comment:
add comment:
`Allocated buffer can be used only for remote access, and can't be accessed by local CPU. In order to obtain the address for remote access, need to call @ref ucp_mem_query and take the field @ref ucp_mem_attr_t.remote_addr`

Reply:
comment updated

Reply:
@hoopoepg ucp.h -> ucpx.h

Reply:
why ucpx.h? it has a strong use case of improving shmem atomics performance